### PR TITLE
keccak: guard against misaligned memory accesses on ARM

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/crypto/keccak.c [hunk 1]
```diff
@@ -104,10 +104,22 @@ void keccak(const uint8_t *in, size_t inlen, uint8_t *md, int mdlen)
     
     memset(st, 0, sizeof(st));
 
-    for ( ; inlen >= rsiz; inlen -= rsiz, in += rsiz) {
+    if (((uintptr_t)in) & 7) { // 64 bit aligned or not ?
+      for ( ; inlen >= rsiz; inlen -= rsiz, in += rsiz) {
+        for (i = 0; i < rsizw; i++) {
+          uint64_t ina;
+          memcpy(&ina, &((uint64_t*)in)[i], 8);
```
Comment (hyc) [inline src/crypto/keccak.c]: shouldn't the source pointer be (void *)? otherwise the compiler will still assume uint64_t alignment, no?

Comment (moneromooo-monero) [reply src/crypto/keccak.c]: OK, I also added a better unit test for this. I can't seem to be able to connect to ssh for now, I'll push later when I can.

**Comment by moneromooo-monero** [comment]
It was 100% identical code indeed. I changed it to only use the memcpy version now.

File: src/crypto/keccak.c (hunk 1 above)
Comment (txchangecoin-project) [reply src/crypto/keccak.c]: Tested and implemented. This will be successful

**Pull request merged**